src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -56,105 +58,72 @@ public class PeriodicBatchingSink : ILogEventSink, IDisposable, IBatchedLogEvent
 
+        _batchedLogEventSink = batchedSink ?? throw new ArgumentNullException(nameof(batchedSink));
         _batchSizeLimit = options.BatchSizeLimit;
-        _queue = new(options.QueueLimit);
-        _status = new(options.Period);
+        _queue = options.QueueLimit is { } limit
+            ? Channel.CreateBounded<LogEvent>(new BoundedChannelOptions(limit) { SingleReader = true })
+            : Channel.CreateUnbounded<LogEvent>(new UnboundedChannelOptions { SingleReader = true });
+        _status = new BatchedConnectionStatus(options.Period);
         _eagerlyEmitFirstEvent = options.EagerlyEmitFirstEvent;
-        _timer = new(_ => OnTick());
+        _delayUntilUnload = Task.Delay(Timeout.InfiniteTimeSpan, _unloading.Token)
+            .ContinueWith(e => e.Exception, TaskContinuationOptions.OnlyOnFaulted);
 
-        // Initialized by externally-callable constructors.
-        _batchedLogEventSink = null!;
+        _loop = Task.Run(LoopAsync, _unloading.Token);
     }
 
     /// <summary>
-    /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+    /// Emit the provided log event to the sink. If the sink is being disposed or
+    /// the app domain unloaded, then the event is ignored.
     /// </summary>
-    /// <filterpriority>2</filterpriority>
-    public void Dispose()
+    /// <param name="logEvent">Log event to emit.</param>
+    /// <exception cref="ArgumentNullException">The event is null.</exception>
+    /// <remarks>
+    /// The sink implements the contract that any events whose Emit() method has
+    /// completed at the time of sink disposal will be flushed (or attempted to,
+    /// depending on app domain state).
+    /// </remarks>
+    public void Emit(LogEvent logEvent)
     {
-        Dispose(true);
-        GC.SuppressFinalize(this);
-    }
+        if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
+
+        if (_unloading.IsCancellationRequested)
+            return;
 
+        _queue.Writer.TryWrite(logEvent);
+    }
+    
     /// <summary>
-    /// Free resources held by the sink.
+    /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
     /// </summary>
-    /// <param name="disposing">If true, called because the object is being disposed; if false,
-    /// the object is being disposed from the finalizer.</param>
-    protected virtual void Dispose(bool disposing)
+    /// <filterpriority>2</filterpriority>
+    public void Dispose()
     {
-        if (!disposing) return;
-
-        lock (_stateLock)
+        try
         {
-            if (_unloading)
-                return;
+            lock (_stateLock)
+            {
+                if (!_unloading.IsCancellationRequested)
+                {
+                    _queue.Writer.Complete();
+                    _unloading.Cancel();
```

Comment:
Perhaps shutdownCts or something might be a better name? It took me till this far down the code despite a decent comment over it above

Reply:
Had a shot at some renames across the board, thanks for the feedback 👍